test/uk/gov/hmrc/merchandiseinbaggage/repositories/DeclarationRepositorySpec.scala

```diff
@@ -85,6 +85,20 @@ class DeclarationRepositorySpec
     }
   }
 
+  "find a declaration by mibReference & Eori" in {
+    val declaration = aDeclaration
+
+    def insert(): Future[Declaration] = repository.insertDeclaration(declaration)
+
+    whenReady(insert()) { insertResult =>
+      insertResult mustBe declaration
+    }
+
+    whenReady(repository.findBy(declaration.mibReference, declaration.eori)) { findResult =>
+      findResult mustBe Some(declaration)
+    }
+  }
+
```

Comment:
Since you're testing a findBy would be good to insert multiple declarations or this test will pass also by calling findAll

Reply:
ok, added inserted two decls